### PR TITLE
lnwallet: update Bolt3 test vectors

### LINKED ISSUE #7407: spec: add new anchor tx test vectors
We should update our test vectors to make sure they match up with this PR: https://github.com/lightning/bolts/pull/1018

### PR DESCRIPTION
**Pull request by ellemouton**
This PR updates adds the `option-zero-fee-htlc-tx` [test vectors](https://github.com/lightning/bolts/pull/1018).
It also adds the new "same amount and preimage" tests that were missing from the existing legacy and anchor tests.

Fixes #7407

### CODE REVIEW

File: lnwallet/test_vectors_anchors.json [hunk 1]
```diff
@@ -11,13 +11,13 @@
     },
     {
```
[the remark this answers is not in the document]
Comment (ellemouton) [reply lnwallet/test_vectors_anchors.json:12]: true! will squash 👍

[the remark this answers is not in the document]
Comment (ellemouton) [reply lnwallet/test_vectors_anchors.json:12]: squashed 🐛

**Comment by ellemouton** [comment]
Just FYI: the [Bolts PR](https://github.com/lightning/bolts/pull/1056) got merged. Can confirm that LDK & Eclair match with these vectors

File: lnwallet/transactions_test.go [hunk 2]
```diff
@@ -186,15 +187,18 @@ func TestCommitmentAndHTLCTransactions(t *testing.T) {
 		err = json.Unmarshal(jsonText, &testCases)
 		require.NoError(t, err)
 
-		t.Run(set.name, func(t *testing.T) {
-			for _, test := range testCases {
-				test := test
+		for _, test := range testCases {
+			test := test
+			name := fmt.Sprintf("%s-%s", set.name, test.Name)
```
Comment (yyforyongyu) [inline lnwallet/transactions_test.go:240]: niiiice

File: lnwallet/transactions_test.go [hunk 3]
```diff
@@ -251,6 +303,18 @@ func addTestHtlcs(t *testing.T, remote,
 func testVectors(t *testing.T, chanType channeldb.ChannelType, test testCase) {
 	tc := newTestContext(t)
 
+	// Determine which htlc set to use.
+	testHtlcs := testHtlcsSet1
+	if strings.Contains(test.Name, "same amount and preimage") {
```
Comment (yyforyongyu) [inline lnwallet/transactions_test.go:308]: 🧐

**Review by yyforyongyu — APPROVED** [review]
LGTM🥳 Thanks for taking care of the format fixes!

File: lnwallet/transactions_test.go (hunk 3 above)
Comment (ellemouton) [reply lnwallet/transactions_test.go:308]: yeah - weird I know 🙃 but the spec does actually define this test like this 🙈

**Pull request merged**